core/src/main/java/org/apache/accumulo/core/conf/AccumuloConfiguration.java

```diff
@@ -33,15 +36,28 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.google.common.base.Preconditions;
 import com.google.common.base.Predicate;
 import com.google.common.base.Predicates;
+import com.google.common.collect.ImmutableMap;
 
 /**
  * A configuration object.
  */
 public abstract class AccumuloConfiguration implements Iterable<Entry<String,String>> {
 
+  private static class PrefixProps {
+    final long updateCount;
+    final Map<String,String> props;
+
+    PrefixProps(Map<String,String> props, long updateCount) {
```

Comment:
If the updateCount is an arbitrary counter, why do you need to continue counting when a new Map is generated? Couldn't you just reset it back to 0?

Reply:
Or couldn't the updateCount instead be a boolean "changed" or "needsRefreshed"?  That is initialized to false and set to true anytime it changes.

Reply:
Using a boolean would require more synchronization.  If a boolean was used syncronization would be needed to prevent the config subclasses from changing while the cache of prefix props was being updated.  Without this sync, changes could be missed if the boolean is set to false after updating the cache but changes occurred concurrently with updating the cache.